### ISSUE TITLE
SVG objects sharing the same css classes

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#5675]
I'm surprised this bug lasted this long. Or maybe i'm missing something, but it seems the following two walls should not share the same CSS classes

files: [20241101 - material and styles.zip](https://github.com/user-attachments/files/17602101/20241101.-.material.and.styles.zip)

![2024-11-01_11-44-37_chromeMY_STOREY_PLAN svg_-_Google_Chromechrome](https://github.com/user-attachments/assets/4bd0e45d-257c-44cd-931f-761da38ef403)

**Comment by sboddy** [comment]
This appears to be resolved to me:
![Image](https://github.com/user-attachments/assets/016d39a3-6216-48bc-8e39-5be9f98ec9fd)

Two materials: top one is Corten, lower one is Concrete.

**Comment by theoryshaw** [comment]
are you getting this weird phenomenon? You can see that one wall has both a steel and concrete texture.  It's actually doubled up, as the video shows.

https://github.com/user-attachments/assets/22c9fa73-ad97-4095-8704-c4e8f08f87fa

**Comment by sboddy** [comment]
I thought you were just being artsy! 😋

Yikes! There are three objects per wall:
![Image](https://github.com/user-attachments/assets/96109154-a6fd-4f94-9543-dd5a6e289721)

![Image](https://github.com/user-attachments/assets/46db0ff6-4124-4a76-aac6-8f49f413d833)

One projection layer with no fill. For some reason **2** cut layers. One looks normal, the other looks weird, adding in multiple matsName and multiple Mats Category entries.

Aha!!!! The problem is that your IfcMaterialLayerSet's are both called "Unnamed". I Changed them ("Marco" & "Polo") and lo and behold:
![Image](https://github.com/user-attachments/assets/42a6492b-8a30-4a48-8b2f-e9595adfdc0c)

Looks like something is relying on naming in a funky way.

**Comment by sboddy** [comment]
A quick look at the new version in Inkscape, we still have three objects, but the two cut objects have the same fill pattern, so it looks OK, although I don't know if this is making edges and aliasing darker by overprinting.

**Comment by sboddy** [comment]
Hey @theoryshaw, can this be closed then?

**Comment by theoryshaw** [comment]
yes, although i'm not sure if this should be caught or not.

**Comment by sboddy** [comment]
Yeah, I've seen a few comments about some "things" in IFC that do not use GUID's, and use the name. Not forcing unique names is a bit of a miss in my opinion. Something like Blenders naming where if a new thing is created, and the name already exists, then it gets a number suffix. Maybe a feature request?

**Comment by theoryshaw** [comment]
I'd like to reopen this issue, as I think it should be a little more robust than this.  That is, I don't think users should be required to put in a IfcMaterialLayerSet.Name as a placeholder just to get this to work.
